presentation/src/main/java/com/smile/presentation/MainViewModel.kt

```diff
@@ -20,19 +19,24 @@ class MainViewModel @Inject constructor(
     private val imageUseCase: UpdateLocalImageUseCase
 ) : BaseViewModel() {
 
-    private val _characterLiveData = MutableLiveData<Wrapper<List<CharacterUI>>>()
-    val characterLiveData: LiveData<Wrapper<List<CharacterUI>>> = _characterLiveData
+    private val _characterLiveData = MutableLiveData<List<CharacterUI>>()
+    val characterLiveData: LiveData<List<CharacterUI>> = _characterLiveData
+    private val liveDataHandler = LiveDataHandler(failure)
 
     init {
         getCharacters()
     }
 
     private fun getCharacters() {
         scope.launch(handler()) {
-            val resp = getCharactersUseCase(Unit).run {
-                CharacterToUIMapper().mapObject(this)
+            val resp = getCharactersUseCase()
+            uiScope.launch {
```

Comment:
Why not just have one type of viewmodel scope (preferably that of `viewmodel-ktx`) and switch between threads using `withContext()`? 🤔

Reply:
I don't know what job is used to create the viewModelScope so I felt it safer to use my predefined scopes so I am in control of how the coroutines behave on exception and how they handle cancelations @chizoba